### PR TITLE
SDI-218: ⬆  Upgrade cypress to v10

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -129,7 +129,7 @@ jobs:
           name: integration tests
           command: npm run int-test
       - store_test_results:
-          path: test-results
+          path: test_results
```
Comment (petergphillips) [inline .circleci/config.yml:132]: left over from a previous change to bring in line with typescript template project that uses `test_results` instead

File: .eslintrc.json [hunk 2]
```diff
@@ -85,7 +85,7 @@
     "comma-dangle": ["error", "always-multiline"],
     "import/no-extraneous-dependencies": [
       "error",
-      { "devDependencies": ["**/*.test.js", "**/*.test.ts", "**/testutils/*.js"] }
+      { "devDependencies": ["**/*.test.js", "**/*.test.ts", "**/testutils/*.js", "cypress.config.ts"] }
```
Comment (petergphillips) [inline .eslintrc.json:88]: was picking up the cypress config here causing the linting to fail as it then preferred cypress over jest

File: cypress.config.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  reporterOptions: {
+    configFile: 'reporter-config.json',
+  },
+  videoUploadOnPasses: false,
+  taskTimeout: 60000,
+  viewportWidth: 1024,
+  viewportHeight: 768,
+  e2e: {
+    setupNodeEvents(on) {
+      on('task', {
```
Comment (petergphillips) [inline cypress.config.ts:24]: plugins/index contents moved here

File: tsconfig.json [hunk 4]
```diff
@@ -19,6 +19,6 @@
     "strictNullChecks": true,
     "typeRoots": ["./server/@types", "./node_modules/@types"]
   },
-  "exclude": ["node_modules", "assets", "integration_tests", "dist"],
+  "exclude": ["node_modules", "assets", "integration_tests", "dist", "cypress.config.ts"],
```
Comment (petergphillips) [inline tsconfig.json:22]: cypress config also needed to be added here to prevent dev dependencies warning

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**